README.md

```diff
@@ -8,43 +8,151 @@ Please create a branch called `homework`. Then, create a file named `homework.js
+
   </li>
     
   <li>Create a function to get the average of a group of numbers 
   <br>
     <code> avg([8, 2, 2, 4]) // should return 4</code>
+    let values = [8, 2, 2, 4];
+let sum = values.reduce((previous, current) => current += previous);
+let avg = sum / values.length;
+
+console.log(values.join(",")+` the avg for this numbers = ${avg}`);
+
   </li>
   <li>Create a function to reverse the letters in a word
   <br>
     <code > reverse("caterpillar") // should return "rallipretac"</code>
+
+
+    function reverseNumber(n)
+{
+	n = n + "";
+	return n.split("").reverse().join("");
+}
+console.log(reverseNumber("caterpillar"));
   </li>
   <li>Create a function that takes an array of words and combines them with a dash
   <br>
     <code > addDashes(['test1', 'test2', 'test3']) // should return "test1-test2-test3"</code>
+
+    function combines(){
+
+  const addDashes= ["test1","test2","test"];
+
+
+  console.log(addDashes.join("-"));
+}
+combines();
   </li>
   <li>Function that will count up to a number and back down and return a string of the climb
   <br>
     <code> countUpAndDown(3) // should return "1 2 3 2 1"</code>
+    function countdown(i) {
+answer = '';
+while( i > 1) {
+answer = answer + i.toString() + ' ';
+i--;
+}
+
+while( i <= 3) {
+answer = answer + i.toString() + ' ';
+i++;
+}
+return answer;
+}
+
+countdown(3);
   </li>
   <li>Write a function that will tell you all of the words in an array that contain the letter `a`
   <br>
     <code > wordsWithA(['cat', 'rabbit', 'dog', 'frog']) // should return ['cat', 'rabbit']</code>
+
+    let wordsWithA =['cat', 'rabbit', 'dog', 'frog'];
+
+function filterItems(arr, query) {
+  return arr.filter(function(el) {
+      return el.toLowerCase().indexOf(query.toLowerCase()) !== -1;
+  })
+}
+
+
+console.log(filterItems(wordsWithA ,'a'));
   </li>
   <li>Write a function that will tell you all of the words in an array that contain a specified letter
   <br>
     <code > wordsWithLetter("g", ['cat', 'rabbit', 'dog', 'frog']) // should return ['dog', 'frog']</code>
+    
+function filterItems(query,arr ) {
+  return arr.filter(function(el) {
+      return el.toLowerCase().indexOf(query.toLowerCase()) !== -1;
+  })
+}
+
+
+console.log(filterItems("g",['cat', 'rabbit', 'dog', 'frog']));
   </li>
   <li>Function that returns the longest word in sentence
   <br>
     <code>longestWord("The cat in the house") // should return "house"</code>
+
+    function longestWo(string) {
+    var str = string.split(" ");
+    var longest = 0;
+    var word = null;
+    str.forEach(function(str) {
+        if (longest < str.length) {
+            longest = str.length;
+            word = str;
+        }
+    });
+    return word;
+}
+console.log(longestWo("The cat in the house"));
   </li>
   <li>Function that returns the largest even number
   <br>
     <code>largestEvenNumber([1,2,3,10,4,7,0]) // should return "10"</code>
+    var array1 = [1,2,3,10,4,7,0];
+
+console.log(Math.max(...array1));
```

Comment:
Please read the question carefully, it states that it wants a **_function_**  that returns the largest **_even_** number.